src/wovnio/wovnphp/Store.php

```diff
@@ -117,14 +118,19 @@ private function updateSettings($updatedOptions)
             sort($this->settings['query'], SORT_STRING);
         }
 
+        if (!empty($this->settings['site_prefix_path'])) {
+            $this->settings['site_prefix_path'] = trim($this->settings['site_prefix_path'], '/');
+        }
+
         // getting the url pattern
         if ($this->settings['url_pattern_name'] === 'query') {
             $this->settings['url_pattern_reg'] = '((\?.*&)|\?)' . $this->settings['lang_param_name'] . '=(?P<lang>[^&]+)(&|$)';
         } elseif ($this->settings['url_pattern_name'] === 'subdomain') {
             $this->settings['url_pattern_reg'] = '^(?P<lang>[^.]+)\.';
         } else {
             $this->settings['url_pattern_name'] = 'path';
-            $this->settings['url_pattern_reg'] = '\/(?P<lang>[^\/.]+)(\/|\?|$)';
+            $prefix = empty($this->settings['site_prefix_path']) ? '' : str_replace('/', '\/', '/' . $this->settings['site_prefix_path']);
+            $this->settings['url_pattern_reg'] = $prefix . '\/(?P<lang>[^\/.]+)(\/|\?|$)';
```

Comment:
I don't really like asserting on url_pattern_reg's regex as this ought to be an implementation detail. Store tests should be checking that we trim the trailing slash etc, but we should test the regex at a higher level - if a URL is processed or not with site prefix path.

Reply:
At StoreTest.php, I just added test of checking `url_pattern_reg` param.
I added this regex's tests at UrlTest.php and UtilsTest.php .
These tests run Store.php internally.

Reply:
That's what I mean. UtilsTest is already testing the outcomes of this regex. It isn't really useful to assert that we are setting a specific regex.

Reply:
I changed tests to not check `url_pattern_reg` param directly.